doc/README.txt

```diff
@@ -1,11 +1,45 @@
 maptlotlib documentation
 ========================
 
+
+Building the documentation
+--------------------------
+
+A list of dependencies can be found in ../doc-requirements.txt.
+
+All of these dependencies can be installed through pip::
+
+  pip install -r ../doc-requirements.txt
+
+or conda::
+
+  conda install sphinx numpydoc ipython mock colorspacious pillow
+
+To build the HTML documentation, type ``python make.py html`` in this
+directory. The top file of the results will be ./build/html/index.html
+
+**Note that Sphinx uses the installed version of the package to build the
+documentation**: matplotlib must be installed *before* the docs can be
+generated. Even if that is the case, one of the files needed to do this,
+'../lib/matplotlib/mpl-data/matplotlibrc', is not version controlled, but
+created when matplotlib is built. This means that the documentation cannot be
+generated immediately after checking out the source code, even if matplotlib
+is installed on your system: you will have to run ``python setup.py build``
+first.
```

Comment:
Not building the finance examples seems fine to me. In the medium term I think we should split the finance module out into it's own package where it can be maintained (or not) by anyone with interest in that. 

For now I suggest moving the finance examples to their own folder. Unless that folder is added to `mpl_example_sections` in https://github.com/matplotlib/matplotlib/blob/master/doc/conf.py they won't be build as part of the docs build. 

Alternatively we can also just add a `# -*- noplot -*-` clause to the first line of the relevant examples.